### DISCUSSION TITLE
Suggest matches from history first

### DISCUSSION THREAD

**Discussion by max-arnold** [minad/vertico#416] [Q&A] [answered]
I never called `sql-mysql` but it is shown on top for some reason. I'd like the recent `my-text-scale-*` commands to be amongst top matches:

![2023-12-03 12 56 00](https://github.com/minad/vertico/assets/370942/934fdeaa-94cc-4f39-b502-701cf72ae90f)

My config:

```elisp

(use-package vertico
  :init
  ;; (setq enable-recursive-minibuffers t)
  (setq vertico-preselect 'first)
  (setq completion-styles '(flex)
        completion-ignore-case t
        read-file-name-completion-ignore-case t
        read-buffer-completion-ignore-case t
        vertico-sort-function #'vertico-sort-history-alpha
        completion-category-defaults nil
        completion-category-overrides '((file (styles basic partial-completion))))
  (vertico-mode)
  (keymap-set vertico-map "RET" #'vertico-directory-enter)
  (keymap-set vertico-map "DEL" #'vertico-directory-delete-char)
  (keymap-set vertico-map "M-DEL" #'vertico-directory-delete-word)
  (add-hook 'rfn-eshadow-update-overlay-hook #'vertico-directory-tidy)
)

(use-package marginalia
  :init
  (marginalia-mode))

```

**Comment by minad** [comment, marked as answer]
The order is overridden by the `flex` completion style here. I suggest you try `substring` or `orderless` instead, which preserve the candidate order. If you want fuzzy matching (while preserving the order) you can configure the `flex` matching style in [Orderless](/oantolin/orderless). Alternatively try out the [Hotfuzz](/axelf4/hotfuzz) completion style, but be aware that Hotfuzz modifies the candidate order.

**Reply by max-arnold** [reply]
Thanks a lot, Daniel!

Orderless did the trick:

```
(setq completion-styles '(orderless)
        ...
        orderless-matching-styles '(orderless-flex))
```
